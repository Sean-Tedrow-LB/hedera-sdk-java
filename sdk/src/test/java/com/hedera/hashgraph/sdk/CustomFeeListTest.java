package com.hedera.hashgraph.sdk;

import io.github.jsonSnapshot.SnapshotMatcher;
import org.junit.AfterClass;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class CustomFeeListTest {
    @BeforeAll
    public static void beforeAll() {
        SnapshotMatcher.start();
    }

    @AfterClass
    public static void afterAll() {
        SnapshotMatcher.validateSnapshots();
    }

    private CustomFeeList spawnCustomFeeListExample() {
        return new CustomFeeList()
            .setCanUpdate(true)
            .addCustomFee(new CustomFixedFee()
                .setFeeCollectorAccountId(new AccountId(4322))
                .setDenominatingTokenId(new TokenId(483902))
                .setAmount(10)
            )
            .addCustomFee(new CustomFractionalFee()
                .setFeeCollectorAccountId(new AccountId(389042))
                .setNumerator(3)
                .setDenominator(7)
                .setMin(3)
                .setMax(100)
            );
    }

    @Test
    void shouldSerialize() {
        assertDoesNotThrow(() -> {
            var originalCustomFeeList = spawnCustomFeeListExample();
            byte[] customFeeListBytes = originalCustomFeeList.toBytes();
            var copyCustomFeeList = CustomFeeList.fromBytes(customFeeListBytes);
            assertTrue(originalCustomFeeList.toString().equals(copyCustomFeeList.toString()));
            SnapshotMatcher.expect(originalCustomFeeList.toString()).toMatchSnapshot();
        });
    }
}