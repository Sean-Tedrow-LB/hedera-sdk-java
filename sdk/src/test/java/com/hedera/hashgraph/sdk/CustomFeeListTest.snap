com.hedera.hashgraph.sdk.CustomFeeListTest.shouldSerialize=[
  "CustomFeeList{canUpdate=true, customFees=[CustomFixedFee{feeCollectorAccountId=0.0.4322, amount=10, demoninatingTokenId=0.0.483902}, CustomFractionalFee{feeCollectorAccountId=0.0.389042, numerator=3, denominator=7, min=3, max=100}]}"
]